app/javascript/shared/mixins/inboxMixin.js

```diff
@@ -36,12 +36,25 @@ export default {
       return this.channelType === INBOX_TYPES.EMAIL;
     },
     isATwilioSMSChannel() {
-      const { phone_number: phoneNumber = '' } = this.inbox;
-      return this.isATwilioChannel && !phoneNumber.startsWith('whatsapp');
+      const { medium: medium = '' } = this.inbox;
+      return this.isATwilioChannel && medium === 'sms';
     },
     isATwilioWhatsappChannel() {
-      const { phone_number: phoneNumber = '' } = this.inbox;
-      return this.isATwilioChannel && phoneNumber.startsWith('whatsapp');
+      const { medium: medium = '' } = this.inbox;
+      return this.isATwilioChannel && medium === 'whatsapp';
+    },
+    findInboxBadgeType() {
+      if (this.isATwilioChannel) {
+        return `${this.isATwilioSMSChannel ? 'sms' : 'whatsapp'}`;
+      }
+      if (this.isATwitterInbox) {
+        return `${
+          this.chat.additional_attributes.type === 'tweet'
+            ? 'twitter-tweet'
+            : 'twitter-chat'
+        }`;
+      }
+      return this.chat.meta.channel;
```

Comment:
`this.chat.meta` might not be available in other places, which might make this mixin unusable @muhsin-k Can you please help @iamsivin to fix the issue?

Reply:
Sure